go/updater/updater.go

```diff
@@ -310,8 +315,18 @@ func (u *Updater) apply(src string, dest string) (tmpPath string, err error) {
 
 // Update checks, downloads and performs an update.
 func (u *Updater) Update(ui libkb.UpdateUI, force bool, requested bool) (update *keybase1.Update, err error) {
+	// If already running, return error
 	u.Lock()
-	defer u.Unlock()
+	if u.running {
+		err = fmt.Errorf("Already running update")
+		u.Unlock()
+		return
+	}
+	u.running = true
+	u.Unlock()
+	defer func() {
+		u.running = false
```

Comment:
I think this [singleflight class](https://github.com/golang/groupcache/blob/master/singleflight/singleflight.go) might be safer.